quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs

```diff
@@ -436,8 +442,8 @@ async fn download_all(
         .object_storage_download_num_bytes
         .clone();
     while let Some(chunk_result) = chunk_stream.next().await {
-        let chunk_response = chunk_result.map_err(AzureErrorWrapper::from)?;
-        let chuck_data = chunk_response.data;
+        let chunk_response = chunk_result?;
```

Comment:
Same here.

Reply:
Why do we do `let chunk_response = chunk_result.map_err(AzureErrorWrapper::from)?;` in one case and `let chunk_response = chunk_result?;` in another?

Reply:
- In this case `let chunk_response = chunk_result.map_err(AzureErrorWrapper::from)?;` we need to propagate `StorageError`, and we cannot jump from `azure::Error` to `StrorageError` we need this map. 

In this case `chunk_response = chunk_result?;` we just need to propagate `AzureErrorWrapper`  

do you think removing it by providing an implementation from `azure::Error` to `StrorageError` makes it better?